test/compilable/issue19243.sh

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/env bash
+
+# bypassing this test:
+#  - on windows
+#  - on FreeBSD 32 bits
+#       test fails looling for liborig.so; don't know why but shouldn't block fixing all other platforms
+#  - on Circle CI with no_pic. (need PIC to run the test)
+if [[ $OS = *"win"* ]]; then exit 0; fi
+if [[ $OS = *"freebsd"* ]] && [[ $MODEL = *"32"* ]]; then exit 0; fi
+if [ ! -z ${CIRCLE_STAGE-} ]; then
+    if [[ ${CIRCLE_STAGE} = *"no_pic"* ]]; then exit 0; fi
+fi
+
+TEST_DIR=${OUTPUT_BASE}
+ORIG_D=$TEST_DIR/orig.d
+ORIG_SO=$TEST_DIR/liborig.so
```

Comment:
Dynamic libraries on macOS has the `.dylib` extension. Probably doesn't really matter.

Reply:
BTW, dynamic libraries are not officially supported on macOS.

Reply:
Should I bypass on OSX too then?
Test might break later

Reply:
Well, if it works it works. That’s why it’s not officially supported because some things work and some won’t.